homeassistant/components/overkiz/siren.py

```diff
@@ -0,0 +1,71 @@
+"""Support for Overkiz sirens."""
+from typing import Any, cast
+
+from pyoverkiz.enums import OverkizState
+from pyoverkiz.enums.command import OverkizCommand, OverkizCommandParam
+
+from homeassistant.components.siren import SirenEntity
+from homeassistant.components.siren.const import (
+    ATTR_DURATION,
+    SUPPORT_DURATION,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HomeAssistantOverkizData
+from .const import DOMAIN
+from .entity import OverkizEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Overkiz sirens from a config entry."""
+    data: HomeAssistantOverkizData = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        OverkizSiren(device.device_url, data.coordinator)
+        for device in data.platforms[Platform.SIREN]
+    )
+
+
+class OverkizSiren(OverkizEntity, SirenEntity):
+    """Representation an Overkiz Siren."""
+
+    _attr_supported_features = SUPPORT_TURN_OFF | SUPPORT_TURN_ON | SUPPORT_DURATION
+
+    @property
+    def is_on(self) -> bool:
+        """Get whether the siren is in on state."""
+        return (
+            self.executor.select_state(OverkizState.CORE_ON_OFF)
+            == OverkizCommandParam.ON
+        )
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Send the on command."""
+        if kwargs.get(ATTR_DURATION):
+            duration = cast(int, kwargs.get(ATTR_DURATION))
```

Comment:
```suggestion
            duration = cast(int, kwargs[ATTR_DURATION])
```

Reply:
If the schema validates you could assert its an `int` instead of casting

Reply:
Looks like its always an `int` if its there https://github.com/home-assistant/core/blob/dev/homeassistant/components/siren/__init__.py#L42

Reply:
Maybe this is better?
```python
            if kw_duration := kwargs.get(ATTR_DURATION):
                duration = cast(int, kw_duration)
```
or
```python
            if kw_duration := kwargs.get(ATTR_DURATION):
                duration = cast(int, kw_duration)
```

Reply:
Addressed in 6ed32982a061eb3878acf9efc4580576e85b5d04, thanks!